gulpfile.js

```diff
@@ -170,19 +215,21 @@ gulp.task('apps', ['dist'], function (done) {
     builder.build(function (err) {
         if (err) {
             console.log("Error building NW apps:" + err);
-            done();
-            return;
+            process.exit(1);
```

Comment:
Should do a `del([appsDir + '**'], { force: true })` before the `process.exit(1)`, or else non-functioning apps will be left behind from the build if it fails.

Reply:
Done, all tasks have a corresponding `clean-*` task that is run before the actual task itself.

Reply:
That is a good thing to do, but it does not solve the problem that still exists:
Because the build tasks run in parallel, at the time the win32 build fails and gulp aborts, there are apps in various half-built stages under `apps/`. These should be deleted, or users will overlook the error, and try to use them, with the same result as trying to use the old osx build.

Reply:
Ofcource, will do a clean task also when an exception is catched.